dln/operator.py

```diff
@@ -146,31 +146,33 @@ def encode(self, string: str) -> List[int]:
     def has_logprobs(self) -> bool:
         return self._has_logprobs
 
+    @staticmethod
+    def _log_filtering_error_message(error_message):
+        error_message = (
+            f"InvalidRequestError, most likely due to "
+            f"content filtering: {error_message}"
+        )
+        logging.warning(error_message)
+        print(colored(error_message, "red"))
+
     @_retry_request(min_wait=4, max_wait=10, max_attempts=100)
     async def _aget_chat_completion_response(self, prompt, **kwargs):
         """
         prompting chatgpt via openai api
         now batching only works for completion, not on chat
         """
         if openai.api_type == "azure":
-            try:
-                response = await openai.ChatCompletion.acreate(
-                    deployment_id=self.engine,
-                    messages=[{"role": "user", "content": prompt}],
-                    **kwargs,
-                )
-            except openai.InvalidRequestError as e:
-                # Most likely a content filtering error from Azure.
-                error_message = f"InvalidRequestError: {e}"
-                logging.warn(error_message)
-                print(colored(error_message, "red"))
-                return str(e)
+            kwargs["deployment_id"] = self.engine
         else:
+            kwargs["model"] = self.engine
+        try:
             response = await openai.ChatCompletion.acreate(
-                model=self.engine,
                 messages=[{"role": "user", "content": prompt}],
                 **kwargs,
             )
+        except openai.InvalidRequestError as e:
+            self._log_filtering_error_message(e)
+            raise
```

Comment:
Does `raise` the same as `raise e`?

Reply:
If it is, then all good. Ready to be merged.

Reply:
Not exactly the same, I changed for `raise e` since this includes the original source